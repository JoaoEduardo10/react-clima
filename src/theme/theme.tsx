import { render } from "@testing-library/react"
import { BrowserRouter, Route, Routes } from "react-router-dom"
import { AppContext } from "../contexts/"
import { GlobalStyle } from "../style/globalStyle"

import { ClimaProps } from "../types/types"

export const renderRoutes = (children: any, path: string) => {
    return render(
        <AppContext>
            <BrowserRouter>
                <Routes>
                    <Route path={path} element={children} />
                </Routes>
                <GlobalStyle />
            </BrowserRouter>
        </AppContext>
    )
}

export const getApi = async (url: string, setAlgo: React.Dispatch<React.SetStateAction<ClimaProps | undefined>>) => {
    const resp = await fetch(url)
    const data = await resp.json()

    setAlgo(data)
}

export const getDate = (data: Date): string[] => {
    const date = data
    const day = date.getDay()
    const month = date.getMonth()

    let newMonth: string = ''
    let newDay: string = ''
    let newDate: string[] = []

    const days = ["Segunda", 'Terça', 'Quarta', 'Quinta', "sexta", "Sabado", "Domingo"]
    const months = ['Janeiro', 'Fevereiro', 'Março', "Abril", 'Maio', "Junho", 'Julho', "Agosto", "Setembro", "Outubo", "Novembro", 'Dezembro']

    

    const numberMonth = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
    const numberDay = [0, 1, 2, 3, 4, 5, 6]

    for(let i = 0; i < days.length; i++) {
        if(day === numberDay[i]){
            newDay = days[i]
            newDate.push(newDay)
            
        }
    }
    

    for(let i = 0; i < months.length; i++) {
        if(month == numberMonth[i]){
            newMonth = months[i]
            newDate.push(newMonth)
            
        }
    }

    return newDate
    
}

export const getHors = (date: Date): string => {
    const horas = date.getHours()
    const minutos = date.getMinutes()

    const newHoras = horas < 10 ? `0${horas}`: horas
    const newMinutos = minutos < 10 ? `0${minutos}` : minutos

    return `${newHoras}:${newMinutos}`
}

export const getTemp = (temp: number, boole: boolean) => {
    if(!boole){
        return `${temp}°C`
    }

    if(boole){
       return `${((temp * 9/5) + 32).toFixed()}°F`
    }
}