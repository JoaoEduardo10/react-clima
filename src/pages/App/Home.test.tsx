import "@testing-library/jest-dom"
import 'jest-styled-components'

import img from "../../img/fundo-clima.jpg";

import Home from './'

import { renderRoutes } from '../../theme/theme'
import { screen } from "@testing-library/react"



describe('<Home />', () => {
    it("should render Home", () => {
        renderRoutes(<Home />, "/")

        const divPai = screen.getByRole('application')

        expect(divPai).toHaveStyleRule('background', `url(${img})`)

    })

    it("should render the home section", () => {
        renderRoutes(<Home />, "/")

        const section = screen.getByRole('heading', { name: "Bem vindo!" }).parentNode

        expect(section).toHaveStyleRule('background-image', "linear-gradient(140deg,#507BA5,#E4ADA7)")
    })

    it("should render button end input", () => {
        renderRoutes(<Home />, "/")

        const inputNome = screen.getByPlaceholderText('Nome')
        const inputCidade = screen.getByPlaceholderText('Cidade')
        const button = screen.getByRole('button')

        expect(inputNome).toBeInTheDocument()
        expect(inputCidade).toBeInTheDocument()
        expect(button).toBeInTheDocument()

    })

    it("should render button end input", () => {
        const { container } = renderRoutes(<Home />, "/")

        expect(container.firstChild).toMatchSnapshot()

    })
})