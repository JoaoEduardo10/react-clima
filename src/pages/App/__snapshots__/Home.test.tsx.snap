// Vitest Snapshot v1

exports[`<Home /> > should render button end input 1`] = `
.c0 {
  background: url(/src/img/fundo-clima.jpg);
  height: 100vh;
  background-size: cover;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  background-image: linear-gradient(140deg,#507BA5,#E4ADA7);
  width: 60vw;
  height: 85%;
  border-radius: .5rem;
}

.c1 h1 {
  margin-top: 7rem;
  margin-bottom: 1rem;
  text-align: center;
  font-size: 6rem;
}

.c1 p {
  margin-bottom: 1rem;
  text-align: center;
  font-size: 1.3rem;
  color: #fff;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 6rem;
}

.c3 {
  margin-bottom: 3rem;
}

.c3 span {
  font-size: 1.8rem;
  color: #363636;
}

.c3 p {
  font-size: 1.5rem;
}

.c3 p.red {
  color: red;
}

.c3 p.withe {
  color: #fff;
}

.c4 {
  display: block;
  width: 38rem;
  margin-top: 1.5rem;
  margin-bottom: .6rem;
  padding: 1.8rem;
  font-size: 1.6rem;
  border: .1rem solid #029b99;
  outline: none;
  border-radius: .5rem;
}

.c4:focus {
  border: .2rem solid #036564;
  background-color: #362D3E;
  color: #fff;
}

.c5 {
  width: 20rem;
  height: 5rem;
  border-radius: 2rem;
  border: 1px solid #AE9DB2;
  box-shadow: .1rem .1rem .5rem #ccc;
  cursor: pointer;
  background-color: #372F3E;
  color: #fff;
  -webkit-transition: all .5s;
  transition: all .5s;
}

.c5:hover {
  background-color: #5D323E;
  color: #fff;
}

@media (max-width:1075px) {
  .c1 form {
    height: 29rem;
  }

  .c1 label {
    margin-bottom: .8rem;
  }

  .c1 label input {
    margin-bottom: .4rem;
  }

  .c1 button {
    height: 40rem;
  }
}

@media (max-width:657px) {
  .c1 {
    width: 95%;
  }

  .c1 h1 {
    font-size: 4.5rem;
    margin-top: 2rem;
  }

  .c1 p.description {
    font-size: 1.2rem;
    text-align: center;
    padding: .5rem 1.4rem;
  }

  .c1 form {
    padding: 0 1rem;
    height: 29.8rem;
    margin-top: 2rem;
  }

  .c1 form p {
    font-size: 1.5rem;
    margin-top: 1rem;
  }

  .c1 form label {
    width: 100%;
    margin-top: 1rem;
  }

  .c1 form label span {
    font-size: 1.4rem;
  }

  .c1 form label input {
    width: 100%;
  }

  .c1 form label button {
    height: 2rem;
  }
}

<div
  class="c0"
  role="application"
>
  <section
    class="c1"
  >
    <h1>
      Bem vindo!
    </h1>
    <p
      class="description"
    >
      Antes de proceguimos digite seu nome e o nome de sua cidade.
    </p>
    <form
      class="c2"
    >
      <label
        class="c3"
      >
        <span>
          Primeiro nome: 
        </span>
        <input
          class="c4"
          placeholder="Nome"
          value=""
        />
      </label>
      <label
        class="c3"
      >
        <span>
          Nome da sua cidade: 
        </span>
        <input
          class="c4"
          placeholder="Cidade"
          value=""
        />
        <p
          class="withe"
        >
          Digite o nome da sua cidade Corretamente
        </p>
      </label>
      <button
        class="c5"
      >
        Come√ßar
      </button>
    </form>
  </section>
</div>
`;
